lib/AppConfig.php

```diff
@@ -118,6 +123,10 @@ public function getAppSettings() {
 				$result[$key] = $value === 'no' ? false : $value;
 			}
 		}
+		if (!empty(self::INTEGER_KEYS[$key])) {
+			$result[$key] = array_map('intval', $result[$key]);
+		}
```

Comment:
This does indeed fix it, the only issue is that the block you added here needs to be added *within* the `foreach` loop, so it can access `$key`. Tried it on my end, and it does resolve it this way.

```suggestion
foreach ($keys as $key) {
    if (str_starts_with($key, 'watermark_')) {
        $value = $this->getAppValueArray($key);
        $value = $value === 'yes' ? true : $value;
        $result[$key] = $value === 'no' ? false : $value;
    }
    // This needs to be inside the foreach loop to get $key
    if (!empty(self::INTEGER_KEYS[$key])) {
        $result[$key] = array_map('intval', $result[$key]);
    }
}
// It used to be here... but then it can't get the $key
```

Reply:
Good catch, thank you. I haven't tested this code yet :point_right: :point_left:

Reply:
tested manually :heavy_check_mark: 
![image](https://github.com/user-attachments/assets/2f3f1fed-c862-4531-9680-bc2548a00ead)